src/util/detect-bugs.js

```diff
@@ -1,23 +1,62 @@
-import TRANSITIONRUN_EVENT_LOOP_BUG from "./detect-bugs/transitionrun-loop.js";
-import UNREGISTERED_TRANSITION_BUG from "./detect-bugs/unregistered-transition.js";
+import detectTransitionRunLoopBug from "./detect-bugs/transitionrun-loop.js";
+import detectUnregisteredTransitionBug from "./detect-bugs/unregistered-transition.js";
 
-/**
- * Data structure for all detected bugs.
- * All bugs start off as true, and once their promises resolve, that is replaced with the actual value
- */
-export const bugs = {
-	TRANSITIONRUN_EVENT_LOOP: true,
-	UNREGISTERED_TRANSITION: true,
-};
+class Bugs {
+	/**
+	 * Whether the browser has the transition run loop bug
+	 * @type {boolean | null}
+	 * @private
+	 */
+	static #transitionRunLoop = null;
 
-TRANSITIONRUN_EVENT_LOOP_BUG.then(value => {
-	bugs.TRANSITIONRUN_EVENT_LOOP = value;
-});
+	/**
+	 * Whether the browser has the unregistered transition bug
+	 * @type {boolean | null}
+	 * @private
+	 */
+	static #unregisteredTransition = null;
 
-UNREGISTERED_TRANSITION_BUG.then(value => {
-	bugs.UNREGISTERED_TRANSITION = value;
-});
+	/**
+	 * Whether the browser has the transition run loop bug
+	 * @type {boolean}
+	 */
+	static get transitionRunLoop () {
+		if (this.#transitionRunLoop === null) {
+			this.#transitionRunLoop = false; // Default to false while detecting
+			detectTransitionRunLoopBug().then(result => {
+				this.#transitionRunLoop = result;
+			});
+		}
+		return this.#transitionRunLoop;
+	}
 
-export { TRANSITIONRUN_EVENT_LOOP_BUG, UNREGISTERED_TRANSITION_BUG };
-export const detected = Promise.all([TRANSITIONRUN_EVENT_LOOP_BUG, UNREGISTERED_TRANSITION_BUG]);
-export default bugs;
+	/**
+	 * Whether the browser has the unregistered transition bug
+	 * @type {boolean}
+	 */
+	static get unregisteredTransition () {
+		if (this.#unregisteredTransition === null) {
+			this.#unregisteredTransition = false; // Default to false while detecting
+			detectUnregisteredTransitionBug().then(result => {
+				this.#unregisteredTransition = result;
+			});
+		}
+		return this.#unregisteredTransition;
+	}
+
+	/**
+	 * Detect all bugs in parallel
+	 * @returns {Promise<void>}
+	 */
+	static async detectAll () {
```

Comment:
If we follow this route, I'd call it `detect`.

Reply:
Good point, done in https://github.com/LeaVerou/style-observer/pull/111/commits/9bc8b023068a242f8423d87fa139d1a88895dc50.